src/hooks/useCourtContracts.js

```diff
@@ -365,11 +368,17 @@ export function useAppealDeposits(disputeId, roundId) {
             const confirmAppealDeposit = nextRound[7]
 
             if (!cancelled) {
-              setAppealDeposits([appealDeposit, confirmAppealDeposit])
+              setAppealDeposits({
+                amounts: [appealDeposit, confirmAppealDeposit],
+                error: null,
+              })
             }
           })
           .catch(err => {
-            console.error(`Error fetching appeal deposits: ${err}`)
+            setAppealDeposits(appealDeposits => ({
+              ...appealDeposits,
+              error: `Error fetching appeal deposits: ${err}`,
```

Comment:
A bit of API nitpick with the errors: in other hooks we're using a boolean value.

We effectively get the same with a string (any error state returned denotes an error if it's truthy), but we may want to consolidate how we design around errors.

Because we are in the same "context" here (i.e. we control both the place where the error is generated and where it'll be consumed), I think it's better to start simpler (ie. prefer bools to long strings). Returning the full error string likely isn't useful to any consumer—if we wanted to log it or send a metric event, we would do so at the error generation site.

In the future, if a consumer needs to branch based on the error type, we could return a short string or symbol.

Reply:
That totally makes sense. Thanks for the explanation.